firedrake/pointeval_utils.py

```diff
@@ -68,7 +66,8 @@ def compile_element(expression, coordinates, parameters=None):
     cell = domain.ufl_cell()
     dim = cell.topological_dimension()
     point = gem.Variable('X', (dim,))
-    point_arg = ast.Decl(utils.ScalarType_c, ast.Symbol('X', rank=(dim,)))
+    # point_arg = ast.Decl(utils.ScalarType_c, ast.Symbol('X', rank=(dim,)))
+    point_arg = lp.GlobalArg("X", dtype=utils.ScalarType_c, shape=(dim,))
```

Comment:
How, if at all, do the changes here (and in `pointquery_utils.py`) effect generated point evaluation kernels (i.e. the kind I know about!)? I'd be keen to see some before/after generated code.